tensorpipe/CMakeLists.txt

```diff
@@ -4,8 +4,16 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
-add_library(tensorpipe
-  ${TP_STATIC_OR_SHARED}
+# List of source files that we need to build libtensorpipe.
+set(TP_PUBLIC_SRCS)
+set(TP_PRIVATE_SRCS)
+set(TP_CUDA_PRIVATE_SRCS)
+
+# List of header files that we need to install.
+set(TP_PUBLIC_HDRS)
+set(TP_CUDA_HDRS)
```

Comment:
Can we call it `TP_PUBLIC_CUDA_HDRS`?

I realize that _right now_ it looks silly to define public headers without defining the private ones, but internally we need to define the private ones too and I would like to keep the definitions aligned. Ideally one day we should find a way to have a single source of truth for both the internal and CMake build systems...

Reply:
done